### ISSUE TITLE
process cannot access the file because it is being used by another process

### ISSUE THREAD

**Issue by 89z** [joshbarrass/deezerdl#4]
Running this command:

~~~
deezerdl download track 75498415 -f MP3_320
~~~

I get this result:

~~~
time="2021-03-03T18:36:51-06:00" level=fatal msg="failed to download track:
rename Maria (after Lyudmila Gurchenko).mp3.enc.part Maria (after Lyudmila
Gurchenko).mp3.enc: The process cannot access the file because it is being used
by another process."
~~~

**Comment by 89z** [comment]
Looks like the error is here:

https://github.com/joshbarrass/deezerdl/blob/07500c8d45d043420732d03ed519088d0c0dc966/internal/downloader.go#L36

With Windows, you have to call `Close` before `Rename`, so you cant `defer` in
this situation. Correct code would look like this:

~~~go
outFile, err := os.Create(outPath + ".part")
if err != nil {
   return err
}
wt := writetracker.NewWriteTracker("")
_, err = io.Copy(outFile, io.TeeReader(resp.Body, wt))
if err != nil {
   return err
}
outFile.Close()
~~~

**Comment by joshbarrass** [comment]
Should be resolved by b8b2a13058ea764d151b2109110367628b5999a6

As you can probably tell by now, I don't use Windows :)

**Comment by 89z** [comment]
Success! I have an audio file finally!
